pyzo/codeeditor/base.py

```diff
@@ -170,89 +169,11 @@ def __init__(self,*args, **kwds):
-
-        S["Syntax.functionname"] = "fore:%s, bold:yes, italic:no, underline:no" % fore3
-        S["Syntax.classname"] = "fore:%s, bold:yes, italic:no, underline:no" % orange
-
-        S["Syntax.string"] = "fore:%s, bold:no, italic:no, underline:no" % violet
-        S["Syntax.unterminatedstring"] = "fore:%s, bold:no, italic:no, underline:dotted" % violet
-        S["Syntax.python.multilinestring"] = "fore:%s, bold:no, italic:no, underline:no" % blue
-
-        S["Syntax.number"] = "fore:%s, bold:no, italic:no, underline:no" % cyan
-        S["Syntax.comment"] = "fore:%s, bold:no, italic:no, underline:no" % yellow
-        S["Syntax.todocomment"] = "fore:%s, bold:no, italic:yes, underline:no" % magenta
-        S["Syntax.python.cellcomment"] = "fore:%s, bold:yes, italic:no, underline:full" % yellow
-
-        S["Editor.Long line indicator"] = "linestyle:solid, fore:%s" % back2
-        S["Editor.Highlight current line"] = "back:%s" % back2
-        S["Editor.Indentation guides"] = "linestyle:solid, fore:%s" % back2
-        S["Editor.Line numbers"] = "back:%s, fore:%s" % (back2, back3)
-
-
-        # Define style using html color names. All 140 legal HTML colour
-        # names can be used (in addition to HEX codes). A full list of
-        # recognized colour names is available e.g. here
-        # http://www.html-color-names.com/color-chart.php
-#         S  = {}
-#         S["Editor.text"] = "back: white, fore: black"
-#         S['Syntax.identifier'] = "fore: black, bold:no, italic:no, underline:no"
-#         S["Syntax.nonidentifier"] = "fore: blue, bold:no, italic:no, underline:no"
-#         S["Syntax.keyword"] = "fore: blue, bold:yes, italic:no, underline:no"
-#         S["Syntax.functionname"] = "fore: black, bold:yes, italic:no, underline:no"
-#         S["Syntax.classname"] = "fore: magenta, bold:yes, italic:no, underline:no"
-#         S["Syntax.string"] = "fore: red, bold:no, italic:no, underline:no"
-#         S["Syntax.unterminatedstring"] = "fore: red, bold:no, italic:no, underline:dotted"
-#         S["Syntax.python.multilinestring"] = "fore: red, bold:no, italic:no, underline:no"
-#         S["Syntax.number"] = "fore: dark orange, bold:no, italic:no, underline:no"
-#         S["Syntax.comment"] ="fore: green, bold:no, italic:yes, underline:no"
-#         S["Syntax.todocomment"] = "fore: magenta, bold:no, italic:yes, underline:no"
-#         S["Syntax.python.cellcomment"] = "fore: green, bold:yes, italic:no, underline:full"
-#         S["Editor.Long line indicator"] = "linestyle:solid, fore: dark grey"
-#         S["Editor.Highlight current line"] = "back: light grey"
-#         S["Editor.Indentation guides"] = "linestyle:solid, fore: light grey"
-#         S["Editor.Line numbers"] = "back: light grey, fore: black"
-
+        
+    ## Style import settings
+        import pyzo
         # Apply style
-        self.setStyle(S)
+        self.setStyle(pyzo.themes[pyzo.config.settings.theme]["data"])
```

Comment:
I kept the default style here. Mostly because this codeeditor submodule was designed to be used separate from Pyzo. So the line above can also not be used. I moved the logic for setting the theme from Pyzo's config to `editor.py`.

Reply:
You're right ! But there's a problem when setting the theme in `editor.py`,  (at startup the theme is not applied correctly on already opened files). The easiest fix is to put the code you wrote :
```suggestion
       # Set style/theme
        try:
            self.setStyle(pyzo.themes[pyzo.config.settings.theme.lower()]['data'])
        except Exception as err:
            print("Could not load theme: " + str(err))
```
in `baseTextCtrl.py`. It also apply the theme to the shell which is nice.

Reply:
Oh, good idea!

Reply:
done!